models/classes/class.QtiTestService.php

```diff
@@ -15,20 +15,21 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2013-2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
- *
+ * Copyright (c) 2013-2023 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```

Comment:
I see we added many refactoring in this file and we have to keep checking code that is not affecting the solution. This makes more difficult to perform the code review.

Can we open a separate PR later for this, if you have time and we are on track with the fix?

Keep in mind we also have to backport it, so less changes the better.

Reply:
Thanks!